routing/missioncontrol.go

```diff
@@ -110,6 +110,10 @@ type MissionControl struct {
 	// results that mission control collects.
 	estimator Estimator
 
+	// onConfigUpdate is a function that is called whenever the
+	// mission control state is updated.
+	onConfigUpdate func(estimator Estimator)
```

Comment:
see my suggestion on the previous review about using `fn.Option` here instead

Reply:
I saw it, on my understanding you proposed using `fn.Option` or a `nil` check, then I have implemented the `nil` check. Maybe I missed somehing.

Reply:
I think `fn.Option` is slightly nicer cause then it never lets us forget the nil check :)

Reply:
I'm doing the following:
`onConfigUpdate fn.Option[func(cfg *MissionControlConfig)]`
and Then:
`m.onConfigUpdate.WhenSome(func(f func(cfg *MissionControlConfig)) {
		f(cfg.Estimator)
})`
But I'm facing a problem to atribute tthe received function at NewMissionControl function:
`OnConfigUpdate func(cfg *MissionControlConfig)`
to the onConfigUpdate, at here:
`	mc := &MissionControl{
		state: newMissionControlState(
			cfg.MinFailureRelaxInterval,
		),
		now:            time.Now,
		selfNode:       self,
		store:          store,
		estimator:      cfg.Estimator,
		onConfigUpdate: cfg.OnConfigUpdate,
`
cannot use cfg.OnConfigUpdate (variable of type func(cfg *MissionControlConfig)) as fn.Option[func(cfg *MissionControlConfig)] value in struct literalcompiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#IncompatibleAssign)